Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -422,7 +422,8 @@ DirectoryBlobReader::DirectoryBlobReader(
     const std::function<void(std::vector<FSTBuilderNode>* fst_nodes)>& sys_callback,
     const std::function<void(std::vector<FSTBuilderNode>* fst_nodes, FSTBuilderNode* dol_node)>&
         fst_callback)
-    : m_encryption_cache(this), m_wrapped_volume(std::move(volume))
+    : m_encryption_cache(this), m_wrapped_volume(std::move(volume)), m_sys_callback(sys_callback),
+      m_fst_callback(fst_callback)
```

Comment:
Storing the callbacks for later use is unsafe, because we have code that calls this constructor with the callbacks set to lambda functions that capture local variables.

Reply:
What is your recommendation? I need to provide something.

Reply:
Make a new constructor that actually copies the data instead of trying to emulate that by re-calling the original one.

Reply:
Are the other reader copy methods acceptable or should I be doing this across the board?

Reply:
The other ones don't have this problem. They're not using callbacks like this one is.

Reply:
Excellent. Working on it now.